### PR TITLE
`merkleDB` -- add inner heap type to syncWorkHeap

### PR DESCRIPTION
**Pull request by danlaine**
## Why this should be merged

Addresses https://github.com/ava-labs/avalanchego/issues/1527. Namely, adds types (rather than `interface{}`) to methods that add/remove items from the `syncWorkHeap` and makes it more clear that the `heap.Interface` methods shouldn't be called directly.

## How this works

Add and use `innerHeap` type inside `syncWorkHeap`. 

## How this was tested

Update existing UT.

### CODE REVIEW

File: x/sync/syncworkheap.go [hunk 1]
```diff
@@ -56,7 +58,7 @@ func (wh *syncWorkHeap) Insert(item *syncWorkItem) {
 		return
 	}
 
-	heap.Push(wh, &heapItem{workItem: item})
+	wh.Push(&heapItem{workItem: item})
```
Comment (danlaine) [inline x/sync/syncworkheap.go:61]: Calls `heap.Push` internally

File: x/sync/syncworkheap.go [hunk 2]
```diff
@@ -65,7 +67,9 @@ func (wh *syncWorkHeap) GetWork() *syncWorkItem {
 	if wh.closed || wh.Len() == 0 {
 		return nil
 	}
-	return heap.Pop(wh).(*heapItem).workItem
+	item := heap.Pop(&wh.innerHeap).(*heapItem)
+	wh.sortedItems.Delete(item)
```
Comment (danlaine) [inline x/sync/syncworkheap.go:71]: `innerHeap.Pop` doesn't call `wh.sortedItems.Delete(item)` so we do it here now

**Review by darioush — APPROVED** [review]

File: x/sync/syncworkheap.go [hunk 3]
```diff
@@ -35,7 +37,7 @@ type syncWorkHeap struct {
 
 func newSyncWorkHeap() *syncWorkHeap {
 	return &syncWorkHeap{
-		priorityHeap: make([]*heapItem, 0),
+		innerHeap: make([]*heapItem, 0),
```
Comment (StephenButtolph) [inline x/sync/syncworkheap.go:40]: I know this isn't part of this PR - but I don't think we need to be doing `make([]T, 0)` here. The default `nil` should be fine.

File: x/sync/syncworkheap.go [hunk 4]
```diff
@@ -128,61 +135,64 @@ func (wh *syncWorkHeap) MergeInsert(item *syncWorkItem) {
...
-	wh.priorityHeap = wh.priorityHeap[:newLength]
+	wh.innerHeap.Swap(newLength, item.heapIndex)
+	wh.innerHeap[newLength] = nil
+	wh.innerHeap = wh.innerHeap[:newLength]
 
 	// the item was already the last item, so nothing needs to be fixed
 	if oldIndex != newLength {
-		heap.Fix(wh, oldIndex)
+		heap.Fix(&wh.innerHeap, oldIndex)
 	}
```
Comment (StephenButtolph) [inline x/sync/syncworkheap.go:161]: We should be using: `heap.Remove` here https://pkg.go.dev/container/heap#Remove.

**Review by StephenButtolph — APPROVED** [review]

**Pull request merged**